project/FrontEnd/collaborative_science_platform/lib/screens/profile_page/profile_page.dart

```diff
@@ -101,238 +113,292 @@ class _ProfilePageState extends State<ProfilePage> {
     }
   }
 
+  void getAuthUser() async {
+    try {
+      final auth = Provider.of<Auth>(context);
+      basicUser = (auth.basicUser ?? {} as BasicUser);
+      isAuthLoading = true;
+      // user = (auth.user ?? {} as User);
+    } catch (e) {
+      setState(() {
+        error = true;
+        errorMessage = "Something went wrong!";
+      });
+      rethrow;
+    } finally {
+      setState(() {
+        isAuthLoading = false;
+      });
+    }
+  }
+
+  void handleButtonIsBanned() {
+    setState(() {
+      isBanned = !isBanned; // Toggle the state for example purposes
+    });
+  }
+
+  void handleButtonIsReviewer() {
+    setState(() {
+      isReviewer = !isReviewer; // Toggle the state for example purposes
+    });
+  }
+
   @override
   Widget build(BuildContext context) {
     final User? user = Provider.of<Auth>(context).user;
     if (user == null) {
-      // guest can see profile pages
+      // TODO guest can see profile pages
```

Comment:
oh I see that it's not a bug but a whole logic change here, need to understand why this is all changed. Could you please briefly summarize?

Reply:
After a discussion, we realized that there was a misunderstanding about these comments. Please delete TODO parts.